federation/pkg/federation-controller/service/dns/dns.go

```diff
@@ -34,18 +34,21 @@ import (
 	fedclientset "k8s.io/kubernetes/federation/client/clientset_generated/federation_clientset"
 	"k8s.io/kubernetes/federation/pkg/dnsprovider"
 	"k8s.io/kubernetes/federation/pkg/dnsprovider/rrstype"
+	"k8s.io/kubernetes/federation/pkg/federation-controller/service/ingress"
 	"k8s.io/kubernetes/federation/pkg/federation-controller/util"
 	"k8s.io/kubernetes/pkg/api/v1"
 	corelisters "k8s.io/kubernetes/pkg/client/listers/core/v1"
 )
 
 const (
-	DNSControllerName = "service-dns"
+	ControllerName = "service-dns"
 
-	DNSUserAgentName = "federation-service-dns-controller"
+	UserAgentName = "federation-service-dns-controller"
 
 	// minDNSTTL is the minimum safe DNS TTL value to use (in seconds).  We use this as the TTL for all DNS records.
 	minDNSTTL = 180
+
+	serviceSyncPeriod = 10 * time.Minute
```

Comment:
yes @quinton-hoole, it could be made shorter. the k8s service controller has a resync period of 30 seconds. Probably we should also can go with 30 secs.